index.d.ts

```diff
@@ -1,5 +1,5 @@
 declare namespace schema {
-  export type SchemaFunction = (value: any, parent: any, key: string) => string
+  export type SchemaFunction = (value: any, parent: any, key: string) => any
```

Comment:
We should have a separate function type for `processStrategy`. `SchemaFunction` is used for `idAttribute`, which should *only* return `string.